crankshaft-engine/src/lib.rs

```diff
@@ -83,3 +108,12 @@ impl Engine {
         });
     }
 }
+
+impl Drop for Engine {
+    fn drop(&mut self) {
+        if let Some(handle) = self.monitoring_handle.take() {
+            debug!("Shutting down monitoring server");
+            handle.abort();
```

Comment:
For a more graceful shutdown, we should have a follow-up PR to pass a cancellation token to the monitoring server (so that it can await on both the cancellation token and its listen socket) which we can simply cancel from this `Drop` implementation.